guides/deployment/fly.md

```diff
@@ -0,0 +1,281 @@
+- The `kill_signal` is set to `SIGTERM`. An Elixir node does a clean shutdown when it receives a `SIGTERM` from the OS.
+
+Some other values were tweaked as well. Check that `internal_port` matches the port for your application.
+
+### Storing secrets on Fly
+
+Before we deploy our new app, we first want to setup a few things in our Fly account. Any secrets that we don't want compiled into the source code are stored externally. An example of that is our Phoenix key base secret.
+
+Elixir has a mix task that generates a new Phoenix key base secret. Let's use that.
+
+```console
+$ mix phx.gen.secret
+REALLY_LONG_SECRET
+```
+
+It generates a long string of random text. Let's store that with Fly as a secret for our app. When we run this command in our project folder, `flyctl` uses the `fly.toml` file to know which app we are setting the value on.
+
+```console
+$ fly secrets set SECRET_KEY_BASE=REALLY_LONG_SECRET
+```
+
+## Provision a database
+
+Most Elixir applications use a database and PostgreSQL is the default one used. Let's provision a database on Fly for our application.
+
+```console
+$ fly postgres create
+```
+
+When naming the database, you can use something like `my-app-db`. Taking the defaults gives you a small database to start playing with.
+
+Now we need to "attach" the database to our application.
+
+```console
+$ fly postgres attach --postgres-app my-app-db
+```
+
+When the database is attached, it creates the secrets needed by your application. You can see what secrets were created this way.
+
+```console
+$ fly secrets list
+```
+
+With our application configured for releases, a Dockerfile defined for packaging it, our `config/runtime.exs` and `rel/env.sh.eex` files configured, a Fly app defined, our secrets stored in Fly, and a database provisioned, we are ready to deploy!
+
+## Deploy time!
+
+Our project is now ready to be deployed to Fly.io.
+
+```console
+$ fly deploy
+```
+
+You can always check on the status of a deploy
+
+```console
+$ fly status
+```
+
+Check your app logs
+
+```console
+$ fly logs
+```
+
+If everything looks good, open your app on Fly
+
+```console
+$ fly open
+```
+
+## Helpful Fly commands and resources
+
+Open the Dashboard for your account
+
+```console
+$ fly dashboard
+```
+
+Scaling your application up or down
+
+```console
+$ fly scale count 2
+```
+
+The [Fly Elixir documentation](https://fly.io/docs/getting-started/elixir) covers things like:
+
+* Connecting to your machine through SSH
+* Getting an IEx shell into a running node
+* Clustering Elixir nodes
+* Scaling to multiple regions while clustered
+
+[Working with Fly applications](https://fly.io/docs/getting-started/working-with-fly-apps/) covers things like:
+
+* Status and logs
+* Custom domains
+* Certificates
+
+## Troubleshooting
+
+See [Troubleshooting](https://fly.io/docs/getting-started/troubleshooting/#welcome-message)
+
+Visit the [Fly Community](https://community.fly.io/) to find solutions and ask questions.
```

Comment:
I would probably add a section on clustering/distribution (i.e. add libcluster with configs xyz).

Reply:
@josevalim I added a sections on:
- getting an IEx shell
- clustering using `libcluster`
- scaling to single region (with clustering)
- scaling to multiple regions (distributed clustering)